chore: inital 13.0.0 blog post (#5403)
electron/electronjs.org: data/blog/electron-13-0.md

---
title: Electron 13.0.0
author:
- VerteDinde
- clavin
- georgexu99
date: '2021-05-25'
---

> Electron 13.0.0 has been released! It includes upgrades to Chromium `91`, V8 `x.x` and Node.js `14.x`. We've added {highlighted changes}, and general improvements. Read below for more details!

Reviewer: ```suggestion
Electron 13.0.0 has been released! It includes upgrades to Chromium `91`, V8 `9.1` and Node.js `14.16.0`. We've added {highlighted changes}, and general improvements. Read below for more details!
```

---

The Electron team is excited to announce the release of Electron 13.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://electronjs.org/releases/stable). Continue reading for details about this release, and please share any feedback you have!

## Notable Changes

### Stack Changes

* Chromium `91`

>     * [New in Chrome 88](https://developer.chrome.com/blog/new-in-chrome-88/)

Reviewer: ```suggestion
    * [New in Chrome 91](https://developer.chrome.com/blog/new-in-chrome-91/)
```

Reviewer: The new in chrome 91 link is not live yet, but pretty sure they're following this URL convention

>     * [New in Chrome 89](https://developer.chrome.com/blog/new-in-chrome-89/)

Reviewer: ```suggestion
    * [New in Chrome 90](https://developer.chrome.com/blog/new-in-chrome-90/)
```

> * Node.js `14.x`

Reviewer: ```suggestion
* Node.js `14.16.0`
```

    * [Node 14.16.0 blog post](https://nodejs.org/en/blog/release/v14.16.0/)
    * [Node 14.0.0 blog post](https://nodejs.org/en/blog/release/v14.0.0/)
* V8 `x.x`

>     * [V8 8.8 blog post](https://v8.dev/blog/v8-release-88)

Reviewer: ```suggestion
    * [V8 9.1 blog post](https://v8.dev/blog/v8-release-91)
```

>     * [V8 8.9 blog post](https://v8.dev/blog/v8-release-89)

Reviewer: ```suggestion
    * [V8 9.0 blog post](https://v8.dev/blog/v8-release-90)
```

---
title: Electron 13.0.0
author:
- sofianguy
- georgexu99
- VerteDinde
date: '2021-05-25'
---

> Electron 13.0.0 has been released! It includes upgrades to Chromium `91`, V8 `9.1` and Node.js `14.16.0`. We've added several API updates, bug fixes, and general improvements. Read below for more details!

Reviewer 2: We're on the same Node.js version right? 
```suggestion
Electron 13.0.0 has been released! It includes upgrades to Chromium `91` and V8 `9.1`. We've added several API updates, bug fixes, and general improvements. Read below for more details!
```

---

The Electron team is excited to announce the release of Electron 13.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://electronjs.org/releases/stable). Continue reading for details about this release, and please share any feedback you have!

## Notable Changes

### Stack Changes

* Chromium `91`
    * [New in Chrome 91](https://www.youtube.com/watch?v=vy6FXa0n1r0)
    * [New in Chrome 90](https://developer.chrome.com/blog/new-in-chrome-90/)

> * Node.js `14.16.0`

Reviewer 3: Actually 14.17.0

Author: It's still 14.16.0 🙂

    * [Node 14.16.0 blog post](https://nodejs.org/en/blog/release/v14.16.0/)
    * [Node 14.0.0 blog post](https://nodejs.org/en/blog/release/v14.0.0/)

> * V8 `x.x`

Reviewer: ```suggestion
* V8 `9.1`
```

* Changed `session.setPermissionCheckHandler(handler)` to allow for `handler`'s first parameter, `webContents` to be `null`. [#19903](https://github.com/electron/electron/pull/19903)

More information about these and future changes can be found on the [Planned Breaking Changes](https://github.com/electron/electron/blob/master/docs/breaking-changes.md) page.

## API Changes

* Added `roundedCorners` option for `BrowserWindow`. [#27572](https://github.com/electron/electron/pull/27572)
* Added new `session.storagePath` API to get the path on disk for session-specific data.[28866](https://github.com/electron/electron/pull/28866)
* Added support for passing DOM elements over the context bridge. [#26776](https://github.com/electron/electron/pull/26776)
* Added `process.uptime()` to sandboxed renderers. [#26684](https://github.com/electron/electron/pull/26684)
* Added missing fields to the parameters emitted as part of the `context-menu `event.[#26788](https://github.com/electron/electron/pull/26788)
* Added support for registering Manifest V3 extension service workers.

> Added ‘registration-completed’ event to ServiceWorkers. [#27562](https://github.com/electron/electron/pull/27562)

Reviewer: ```suggestion
* Added ‘registration-completed’ event to ServiceWorkers. [#27562](https://github.com/electron/electron/pull/27562)

```

* Added ‘registration-completed’ event to ServiceWorkers. [#27562](https://github.com/electron/electron/pull/27562)

### Removed/Deprecated Changes

The following APIs have been removed or are now deprecated:

* Deprecated the `new-window` event of `WebContents`. It is replaced by `webContents.setWindowOpenHandler()`
* Removed deprecated `shell.moveItemToTrash()`. [#26723](https://github.com/electron/electron/pull/26723)
* Removed the following deprecated `BrowserWindow` extension APIs:

    * `BrowserWindow.addExtension(path)`
    * `BrowserWindow.addDevToolsExtension(path)`
    * `BrowserWindow.removeExtension(name)`
    * `BrowserWindow.removeDevToolsExtension(name)`
    * `BrowserWindow.getExtensions()`
    * `BrowserWindow.getDevToolsExtensions()`

>     Use the session APIs instead:

Reviewer 2: ```suggestion
    Use the `session` APIs instead:
```

    * `ses.loadExtension(path)`
    * `ses.removeExtension(extension_id)`
    * `ses.getAllExtensions()`

> * The following systemPreferences methods have been deprecated:

Reviewer 2: ```suggestion
* The following `systemPreferences` methods have been deprecated:
```

    * `systemPreferences.isDarkMode()`
    * `systemPreferences.isInvertedColorScheme()`
    * `systemPreferences.isHighContrastColorScheme()`

>     Use the following nativeTheme properties instead:

Reviewer 2: ```suggestion
    Use the following `nativeTheme` properties instead:
```

---
title: Electron 13.0.0
author:
- sofianguy
- georgexu99
- VerteDinde
date: '2021-05-25'
---

Electron 13.0.0 has been released! It includes upgrades to Chromium `91`, V8 `9.1` and Node.js `14.16.0`. We've added several API updates, bug fixes, and general improvements. Read below for more details!

---

The Electron team is excited to announce the release of Electron 13.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://electronjs.org/releases/stable). Continue reading for details about this release, and please share any feedback you have!

> ## Notable Changes

Reviewer 4: I know we haven't fully figured it out yet (at least the electron-rebuild part) but do we want to mention the native module thing here? Depends on whether someone can figure out the electron-rebuild thing today I guess.

Author: Working on getting the electron-rebuild part working, I'll add the native module fix here, that's a good callout 🙂

Author: Still getting this up here: https://github.com/electron/electron-rebuild/pull/703 I'll leave out of the blog post for now and add in when we add the 91 blog post.

### Stack Changes

* Chromium `91`
    * [New in Chrome 91](https://developer.chrome.com/blog/new-in-chrome-91/)
    * [New in Chrome 90](https://developer.chrome.com/blog/new-in-chrome-90/)
* Node.js `14.16.0`
    * [Node 14.16.0 blog post](https://nodejs.org/en/blog/release/v14.16.0/)
    * [Node 14.0.0 blog post](https://nodejs.org/en/blog/release/v14.0.0/)
* V8 `9.1`
    * [V8 9.1 blog post](https://v8.dev/blog/v8-release-91)
    * [V8 9.0 blog post](https://v8.dev/blog/v8-release-90)

### Highlight Features

* Added `process.contextIsolated` property that indicates whether the current renderer context has `contextIsolation` enabled. [#28252](https://github.com/electron/electron/pull/28252) 
* Added new `session.storagePath` API to get the path on disk for session-specific data. [#28866](https://github.com/electron/electron/pull/28866) 
* Deprecated the `new-window` event of `WebContents`. It is replaced by `webContents.setWindowOpenHandler()`

> * The `submitURL` option for `crashReporter.start` is no longer a required argument when `uploadToServer` is false. [#28283](https://github.com/electron/electron/pull/28283)

Reviewer 4: This one feels like more of a bugfix and not up at the same level as the other items in this section?

Author: Yeah, true - I'll remove this one
